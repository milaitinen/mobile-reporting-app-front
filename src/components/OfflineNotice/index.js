import OfflineNotice from './OfflineNotice';

export { OfflineNotice };