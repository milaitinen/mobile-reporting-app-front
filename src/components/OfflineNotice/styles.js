import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({

    offlineContainer: {
        backgroundColor: '#b52424',
        height:30,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row',
        position: 'absolute',
        top: 15,
        alignSelf: 'stretch',
    },

    offlineText: {
        color: '#fff',
    },


});