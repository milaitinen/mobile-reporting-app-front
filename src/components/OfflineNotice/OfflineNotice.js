import React from 'react';
import { View, StatusBar, Platform } from 'react-native';
import styles from './styles';
import { connect } from 'react-redux';

/**
 * A modified status bar -component that changes color depending on the network connection status
 */
export class OfflineNotice extends React.Component {

    render() {
        const backgroundColor = this.props.isConnected ? '#3d4f7c' : '#b52424';

        if (Platform.OS === 'ios') {
            return (
                <View style={[styles.offlineContainer, { backgroundColor }]}>
                    <StatusBar translucent backgroundColor={ backgroundColor }/>
                </View>
            );
        } else {
            return (
                <StatusBar backgroundColor={ backgroundColor } barStyle="light-content" />
            );
        }
    }
}

export const mapStateToProps = ( state ) => { // exported for testing purposes
    return {
        isConnected: state.connection.isConnected,
    };
};

export default connect(mapStateToProps)(OfflineNotice);
