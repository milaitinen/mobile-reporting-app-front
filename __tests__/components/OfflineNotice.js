import 'react-native';
import { Platform } from 'react-native';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { OfflineNotice } from '../../src/components/OfflineNotice/OfflineNotice';
import { mapStateToProps } from '../../src/components/OfflineNotice/OfflineNotice';

configure({ adapter: new Adapter() });
jest.mock('StatusBar', () => 'StatusBar');

describe('Offline notice component', () => {

    it('renders correctly for android', () => {
        Platform.OS = 'android';
        const tree = renderer.create(
            <OfflineNotice isConnected={true} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly for IOS', () => {
        Platform.OS = 'ios';
        const tree = renderer.create(
            <OfflineNotice isConnected={true} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly without a connection', () => {
        const tree = renderer.create(
            <OfflineNotice isConnected={false} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should show connected value', () => {
        const initialState = {
            connection: {
                isConnected: true
            }
        };
        expect(mapStateToProps(initialState).isConnected).toEqual(true);
    });
});

