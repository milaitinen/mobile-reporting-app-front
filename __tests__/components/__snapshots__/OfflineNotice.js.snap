// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offline notice component renders correctly for IOS 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#3d4f7c",
      },
    ]
  }
>
  <StatusBar
    backgroundColor="#3d4f7c"
    translucent={true}
  />
</View>
`;

exports[`Offline notice component renders correctly for android 1`] = `
<StatusBar
  backgroundColor="#3d4f7c"
  barStyle="light-content"
/>
`;

exports[`Offline notice component renders correctly without a connection 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#b52424",
      },
    ]
  }
>
  <StatusBar
    backgroundColor="#b52424"
    translucent={true}
  />
</View>
`;
